File: code/leetcode/238 ProductOfArrayExceptSelf/ProductOfArrayExceptSelf.js
var productExceptSelf = function(nums) {
    let length = nums.length;
    let result = [1];
    let temp = 1;
    for (let i = 1; i < length; i++) {
        result[i] = result[i - 1] * nums[i - 1];
    };
    for (let i = length - 1; i > -1; i--) {
        result[i] = result[i] * temp;
        temp = temp * nums[i];
    };
    return result;
};

var nums=[1,2,3,4];

console.log(productExceptSelf(nums))

File: code/leetcode/238 ProductOfArrayExceptSelf/ProductOfArrayExceptSelf.java


class Solution {
    public int mul(int []a,int i){
        int pro=1;
        //    boolean isZeroPresent = false;
        for(int j=0;j<a.length;j++){
            if(j!=i)
                pro*=a[j];
        }
        System.out.println("called");
        return pro;

    }
    public int[] productExceptSelf(int[] nums) {

        boolean isZeroPresent = false;
        int ans[] = new int[nums.length];

        for(int i=0;i<nums.length;i++){

            ans[i]= mul(nums,i);
        }


        return ans;
    }
}

public class ProductOfArrayExceptSelf {
    public static void main(String[] args) {
        System.out.println("heelo");
        int ar[] = {-1,1,0,-3,3};
        Solution sol = new Solution();
        int ans[] = sol.productExceptSelf(ar);
        for(int i=0;i<ans.length;i++)
            System.out.println(ans[i]);

    }

}
